src/rules/selector-pseudo-class-parentheses-space-inside/index.js

```diff
@@ -0,0 +1,84 @@
+import _ from "lodash"
+import selectorParser from "postcss-selector-parser"
+import {
+  report,
+  ruleMessages,
+  styleSearch,
+  validateOptions,
+} from "../../utils"
+
+export const ruleName = "selector-pseudo-class-parentheses-space-inside"
+
+export const messages = ruleMessages(ruleName, {
+  expectedOpening: "Expected single space after \"(\"",
+  rejectedOpening: "Unexpected whitespace after \"(\"",
+  expectedClosing: "Expected single space before \")\"",
+  rejectedClosing: "Unexpected whitespace before \")\"",
+})
+
+export default function (expectation) {
+  return (root, result) => {
+    const validOptions = validateOptions(result, ruleName, {
+      actual: expectation,
+      possible: [
+        "always",
+        "never",
+      ],
+    })
+    if (!validOptions) { return }
+
+    root.walkRules(rule => {
+      selectorParser(selectorTree => {
```

Comment:
Let's also do an optimization here, by not parsing the selector if there is no `(` within it.